torch/csrc/distributed/rpc/utils.cpp

```diff
@@ -353,6 +353,105 @@ std::pair<std::vector<char>, std::vector<at::Tensor>> wireDeserialize(
   return {std::move(payload), std::move(tensors)};
 }
 
+TensorPipeEntry tensorpipeSerialize(const Message& rpcMessage) {
+  tensorpipe::Message tpMessage;
+  std::vector<torch::Tensor> reservedTensors;
+  std::vector<std::vector<uint8_t>> copiedTensors;
+
+  const std::vector<char>& payload = rpcMessage.payload();
+  c10::List<at::Tensor> tensors = cloneSparseTensors(rpcMessage.tensors());
+
+  // Payload
+  tpMessage.data = (uint8_t*)(payload.data());
+  tpMessage.length = payload.size();
+
+  // Metadata - encode rpc message type + message id
+  tpMessage.metadata = c10::str(
+      static_cast<int>(rpcMessage.type()),
+      ":",
+      static_cast<int>(rpcMessage.id()));
```

Comment:
Can we avoid doing this since type() and id() have fixed size? Just make the first 4 bytes of metadata type() and next 4 bytes id()?

Reply:
The code here is wrong, since it truncates off the top 32 bits off the rpc::Message's int64_t.

I agree with Pritam that the string parsing might be a little heavyweight. (If the rest of the message were text, I'd lean towards a text encoding, but here, it seems like binary might be nice)

The main caveats are:
 - make sure to leave 8 bytes for the id()
 -  don't know if we care about byte-order here? it might be good form to htonl/etc if you want, though basically everybody uses arm/x86 these days :)

Reply:
Regarding byte-order I used `THP_encodeInt64Buffer` here https://github.com/pytorch/pytorch/blob/master/torch/csrc/distributed/autograd/rpc_messages/rpc_with_autograd.cpp#L80, we can do something similar for this.

Reply:
> The code here is wrong, since it truncates off the top 32 bits off the rpc::Message's int64_t.

Good catch!